kratos/integration/integration_point_utilities.cpp

```diff
@@ -42,6 +56,44 @@ namespace Kratos
         }
     }
 
+    /* @brief Creates integration points within provided spans according
+     *        to the provided points per span.
+     * @param rIntegrationPoints result integration points.
+     * @param rSpansLocalSpace spans in which integration points are generated.
+     * @param IntegrationPointsPerSpan number of integration points per span.
+     */
+    void IntegrationPointUtilities::CreateIntegrationPoints1DGrid(
+        IntegrationPointsArrayType& rIntegrationPoints,
+        const std::vector<double>& rSpansLocalSpace,
+        const SizeType IntegrationPointsPerSpan)
+    {
+        const SizeType num_spans = rSpansLocalSpace.size() - 1;
+        const SizeType number_of_integration_points = num_spans * IntegrationPointsPerSpan + num_spans + 2;
```

Comment:
why + num_spans + 2?

Reply:
So you have two additional for either end, but why +num_spans?

Reply:
yeah.... that was actually wrong, the correct value is `+1`. Basically, the idea is to have points at the beginning and end of spans. Then via the provided `IntegrationPointsPerSpan`, the middle is filled up. Accordingly, by having `IntegrationPointsPerSpan=0` we will have only the span locations, which is: `num_spans + 1`. For all `IntegrationPointsPerSpan bigger than 0` it adds up `IntegrationPointsPerSpan * num_spans`